104.md

```diff
@@ -0,0 +1,205 @@
+
+## New Event Kinds
+
+This NIP introduces three new event kinds. First, we'll look at the two that are related to the conversation itself. These event kinds MUST always be gift-wrapped as per [NIP-59](https://github.com/nostr-protocol/nips/blob/master/59.md) and use [NIP-44](https://github.com/nostr-protocol/nips/blob/master/44.md) encryption for the [Seal (kind 13)](https://github.com/nostr-protocol/nips/blob/master/59.md#2-the-seal-event-kind) and [Gift Wrap (kind 1059)](https://github.com/nostr-protocol/nips/blob/master/59.md#3-gift-wrap-event-kind) events.
+
+This combination might seem like overkill given we're further encrypting the actual content of the messages between the users using a double-ratchet symmetrical encryption scheme but this combination significantly decreases the amount of metadata published and improves deniability/repudiation of the conversation.
+
+### Conversation Request Event
+
+This event is sent when one user wants to start a conversation with another user. For example, if Alice wants to talk to Bob, her first message will be a conversation request event. This message allows both parties to establish a shared root key via a Diffie-Hellman key exchange and allows Alice to derive the first key in the chain to encrypt a message. An encrypted message MUST be included in the conversation request event to ensure that the first ratchet step is performed. The message can be a default message like, "Hi, I would like to connect." if no message is provided by the user.
+
+```json
+{
+    "kind": 443,
+    "created_at": <unix timestamp in seconds>,
+    "pubkey": <pubkey of the sender>,
+    "content": <ciphertext of initial message symmetrically encyrpted using first sending chain/message key>,
+    "tags": [
+        ["p", <pubkey of the recipient>],
+        ["prekey", <pubkey of the recipient prekey used for DH calculation>],
+        ["ephemeral", <pubkey of the ephemeral key used for DH calculation>]
+    ],
+    "sig": "" // No signature because these are only sent inside gift wraps.
+}
+```
+
+### Double Ratchet Direct Message (DRDM) Event
+
+The Double Ratchet Direct Message (DRDM) event is the main event kind that carries the conversation content. This event carries enough information for recievers to properly double ratchet their chains and decrypt the message content with the right symmetrical keys.
+
+```json
+{
+    "kind": 444,
+    "created_at": <unix timestamp in seconds>,
+    "pubkey": <pubkey of the sender>,
+    "content": <ciphertext of the message symmetrically encrypted with chain/message key>,
+    "tags": [
+        ["p", <pubkey of the recipient>],
+        ["dh_sending", <current DH sending pubkey>],
+        ["current_index", <number of current message in the current message chain>],
+        ["previous_length", <length of previous message chain>]
+    ],
+    "sig": "" // No signature because these are only sent inside gift wraps.
+}
+```
+
```

Comment:
Clients MUST check if the `pubkey` of kind `443` and `444` is the same as the Seal's, kind `13`, pubkey. Otherwise, impersonators can impersonate. :)